llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -121,8 +121,9 @@ static bool sinkScalarOperands(VPlan &Plan) {
       continue;
     for (auto &Recipe : *VPBB) {
       for (VPValue *Op : Recipe.operands())
-        if (auto *Def =
-                dyn_cast_or_null<VPSingleDefRecipe>(Op->getDefiningRecipe()))
+        // TODO: Add support for multiple-def recipes.
+        if (auto *Def = Op->getDefiningRecipe();
+            Def && Def->getNumDefinedValues() == 1)
```

Comment:
This transform (`sinkScalarOperands()`) is yet to be updated to support multiple defs.